### DISCUSSION TITLE
A simple project turned frustrating

### DISCUSSION THREAD

**Discussion by Starzs906** [python-telegram-bot/python-telegram-bot#3690] [Q&A]
I recently followed a guide provided by Telegram in Java. 

The goal of the project was simple, to send a message to a Telegram user AND to be able to clearly receive messages from the Telegram user, no automatic responses or sending emojis or pictures, just text.

I am looking to do repeat the same project in python.

Here is where I am getting caught up; the following code gives me the exact output I want when receiving messages from a Telegram User, how do I replicate this ?

    public void onUpdateReceived(Update update) {
        var msg = update.getMessage();
        var user = msg.getFrom();
        System.out.println(user.getFirstName() + " wrote " + msg.getText());

I have seen tons of tutorials online but they are all on automatic replies or a message handler function, not simply printing the message in output.

**Comment by y3boy** [comment]
Hi.
To get recieve message U can use code above.
```
import asyncio
from telegram import Update
from telegram.ext import Application, CommandHandler, ContextTypes, MessageHandler, filters


async def start(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    await update.message.reply_text(f"Hi {update.effective_user.username}!")


async def echo_to_stdout(update: Update, context: ContextTypes.DEFAULT_TYPE) -> None:
    print(f'{update.effective_user.username} + " wrote: " + {update.message.text}', flush=True)
    # "Empty" await func
    await asyncio.sleep(0.01)


def main() -> None:
    application = Application.builder().token("TOKEN:XXXXXXXX").build()
    application.add_handler(CommandHandler("start", start))
    application.add_handler(MessageHandler(filters.TEXT & ~filters.COMMAND, echo_to_stdout))
    application.run_polling()


if __name__ == "__main__":
    main()
```
